tests/ui/needless_borrow.rs

```diff
@@ -251,3 +251,10 @@ mod issue_10253 {
         (&S).f::<()>();
     }
 }
+
+fn issue_12268() {
+    let option = Some((&1,));
+    let x = (&1,);
+    // Lint here.
+    option.unwrap_or((&x.0,));
```

Comment:
It's generally better to use tagging to indicate that lints should be present. This helps prevent regressions.
```suggestion

    option.unwrap_or((&x.0,));
    //~^ ERROR: this expression creates a reference which
```

For more details on tagging:
https://doc.rust-lang.org/clippy/development/writing_tests.html#write-test-cases